### PR TITLE
Maven options to make build faster

### PR DESCRIPTION
**Pull request by aziemchawdhary-gs**
This change makes build workflow consistent with release workflow

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -86,7 +88,9 @@ jobs:
         # and can cause problem with some code generators
         # See https://github.com/finos/legend-engine/pull/924
         run: |
+          export MAVEN_OPTS="-Xmx12g"
           mvn -B -e -DskipTests=true deploy -P docker-snapshot,sonar
+          export MAVEN_OPTS="-Xmx2g"
```
[the remark this answers is not in the document]
Comment (aziemchawdhary-gs) [reply .github/workflows/build.yml:93]: Surefire will spin out a separate process for which we set 70% of host memory. The main process doesn't need much so this makes more available for the test process. The second maven invocation is reusing the artefacts from the first maven invocation, so we don't need heap to build anything.

**Comment by rafaelbey** [comment]
The change make sense to me.  Why we cancelled the build job?

**Comment by finos-admin** [comment]
This PR is stale because it has been open for 30 days with no activity. Please remove stale label or add any comment to keep this open. Otherwise this will be closed in 5 days.

**Comment by finos-admin** [comment]
This PR was closed because it has been inactive for 35 days. Please re-open if this PR is still relevant.

**Pull request closed without merging**